symbolic-debuginfo/src/dwarf.rs

```diff
@@ -24,8 +24,24 @@ use thiserror::Error;
 use symbolic_common::{AsSelf, Language, Name, NameMangling, SelfCell};
 
 use crate::base::*;
+#[cfg(feature = "macho")]
 use crate::macho::BcSymbolMap;
-use crate::private::FunctionStack;
+use crate::shared::FunctionStack;
+
+/// This is a fake BcSymbolMap used when macho support is turned off since they are unfortunately
+/// part of the dwarf interface
```

Comment:
maybe the api of BcSumbolMap was a poor choice and personally i'd be open to changing it if it results in something better.

The toplevel API is `DwarfDebugSession::load_symbolmap()` I think, which you already made conditional to `macho`.  I think if you might be able to remove the `bcsymbolmap` arg from `UnitRef::resolve_function_name` and instead put it as a field in the `UnitRef` struct if `macho` is enabled and then it can probably be made to work with some more conditional code.

This way `BcSymbolMap` could completely disappear from the public API if `macho` is not enabled?  What do you think?

Reply:
I initially started on that route, but it seemed like I was changing a bit too much and figured such a change could be better left for later.